clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -89,18 +89,26 @@ public void append(long offset, Record record) {
     /**
      * Append a new record and offset to the buffer
      */
-    public void append(long offset, long timestamp, byte[] key, byte[] value) {
+    public void append(long offset, long timestamp, long checksum, byte[] key, byte[] value) {
         if (!writable)
             throw new IllegalStateException("Memory records is not writable");
 
         int size = Record.recordSize(key, value);
         compressor.putLong(offset);
         compressor.putInt(size);
-        compressor.putRecord(timestamp, key, value);
+        compressor.putRecord(timestamp, checksum, key, value);
         compressor.recordWritten(size + Records.LOG_OVERHEAD);
     }
 
     /**
+     * Append a new record and offset to the buffer
+     */
+    public void append(long offset, long timestamp, byte[] key, byte[] value) {
+        long crc = Record.computeChecksum(timestamp, key, value, CompressionType.NONE, 0, -1);
```

Comment:
If you look at original implementation:
the above append() method was calling compressor.putRecord which took timestamp, key, value parameters. See Compressor.java, line 212 above. That method looked like this:
public void putRecord(long timestamp, byte[] key, byte[] value) {		
  putRecord(timestamp, key, value, CompressionType.NONE, 0, -1);		
}

Basically, append path is to append an inner message which always have compression type == none (otherwise, broker will think that inner message is  another compressed message containing a bunch of other messages).